Support Poetry Config (#3)
MaterializeInc/dbt-action: README.md

| Name                 | Description                                  | Default       |
| -------------------- | -------------------------------------------- | ------------- |
| `DBT_MZ_VERSION`     | `dbt-materialize` adapter version to install | `latest`      |

> | `POETRY_VERSION`     | Poetry version if dbt project is managed with poetry. Cannot be used with `DBT_MZ_VERSION` | |

Reviewer: ```suggestion
| `POETRY_VERSION`     | Poetry version, if the dbt project is managed with [Poetry](https://python-poetry.org/). Exclusive with `DBT_MZ_VERSION`. | |
```

Reviewer: Wouldn't the Poetry version also be available in the user's `pyproject.toml`? Sounds a bit like a chicken and egg problem. I'm wondering if we could go from that file and `pip install .` instead of requiring users to set the Poetry version on the side. IIUC, this would not install dev dependencies by default (see [Poetry #3514](https://github.com/python-poetry/poetry/issues/3514)), but dependency management seems to have changed quite a bit in [1.2.0](https://python-poetry.org/blog/announcing-poetry-1.2.0/#dependency-groups).
